docs/site/submitting_a_pr.md

```diff
@@ -0,0 +1,629 @@
+
+To help with abiding by the rules of commit messages, please use the `commitizen` tool mentioned in the documentation above. This means will we use `git cz` instead of `git commit`.
+
+In my case, I want the commit message to look like this:
+
+```
+docs: tutorial on how to submit a pull request to lb4
+
+This tutorial shows contributors step-by-step instructions on how to submit a pull request (PR) to Loopback v4
+
+```
+
+If you need the long description to span multiple lines, use `\n` characters to separate the sentences.
+
+Let's go through the interactive prompts of `commitizen`.
+
+Type
+```
+git cz
+```
+
+The first prompt asks you to choose the `type` value:
+
+![submit_pr_git_cz_1.png](./imgs/submit_pr_git_cz_1.png)
+
+
+Using the `down arrow` key, I will choose `docs` and press `ENTER`.
+
+![submit_pr_git_cz_2.png](./imgs/submit_pr_git_cz_2.png)
+
+The next prompt asks you to choose the `scope` value:
+
+![submit_pr_git_cz_3.png](./imgs/submit_pr_git_cz_3.png)
+
+In my case, I won't be specifying a scope value, so I will simply press `ENTER` to skip.
+
+The next prompt asks you to specify the `short description` value:
+
+![submit_pr_git_cz_4.png](./imgs/submit_pr_git_cz_4.png)
+
+I specify : `tutorial on how to submit a pull request to lb4` .
+
+![submit_pr_git_cz_5.png](./imgs/submit_pr_git_cz_5.png)
+
+The next prompt asks you to specify the `long description` value:
+
+![submit_pr_git_cz_6.png](./imgs/submit_pr_git_cz_6.png)
+
+I specify : `This tutorial shows contributors step-by-step instructions on how to submit a pull request (PR) to Loopback v4`
+
+![submit_pr_git_cz_7.png](./imgs/submit_pr_git_cz_7.png)
+
+
+The next prompt asks if any `breaking changes` will be introduced.
+
+![submit_pr_git_cz_8.png](./imgs/submit_pr_git_cz_8.png)
+
+I specify : `N`
+
+![submit_pr_git_cz_9.png](./imgs/submit_pr_git_cz_9.png)
+
+The next prompt asks if this change affects any open issues.
+
+![submit_pr_git_cz_10.png](./imgs/submit_pr_git_cz_10.png)
+
+I specify : `N`
+
+![submit_pr_git_cz_11.png](./imgs/submit_pr_git_cz_11.png)
+
+The interactive `commitizen` prompts complete and the commit is created with a properly formatted message.
+
+![submit_pr_git_cz_12.png](./imgs/submit_pr_git_cz_12.png)
+
+To view the commit message that `commitizen` has created, you can type :
+
+```
+git log
+```
+
+![submit_pr_git_cz_13.png](./imgs/submit_pr_git_cz_13.png)
+
+Press `q` to exit the list of commits.
+
+
+### 14. Push commits to your remote feature branch
+
+It is now time to push your local committed changes from your local feature branch directory to your remote feature branch; to keep them in synch.
+
+Type :
+
+```
+git push
+```
+
+![submit_pr_git_push_1.png](./imgs/submit_pr_git_push_1.png)
+
+Open your browser, and navigate to your forked repository, and you will notice the commit.
+
+![submit_pr_git_push_2.png](./imgs/submit_pr_git_push_2.png)
+
+
+### 15. Rebasing
```

Comment:
I think for step 15 and 16, we should create a separate section for that. 
"What if your PR gets out of date"?
"What if you have multiple commits and want to squash them into one?"

Not all PRs need to go through those 2 steps but in case contributors need to do it, they can follow the instructions there.